File: bank-service/src/main/java/rs/edu/raf/model/ExchangeAccount.java
package rs.edu.raf.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ExchangeAccount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private BigDecimal balance;
    private String currencyCode;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/ExchangeAccount.kt
package org.example.bankservicekotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.math.BigDecimal

@Entity
class ExchangeAccount(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long,
    var balance: BigDecimal,
    val currencyCode: String) {
}